File: src/main/java/com/menevseoglu/okan/twitterhoaxfinder/request/HoaxFinderSearchRequest.java
package com.menevseoglu.okan.twitterhoaxfinder.request;

import lombok.Data;
import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.NotEmpty;

@Data
public class HoaxFinderSearchRequest {

    @NotEmpty
    @Length(max = 280)
    private String query;

    private String screenName;

    private String lang;
}

File: twitter-hoax-finder-client/src/app/search/shared/hoax-finder-search-request.ts
export class HoaxFinderSearchRequest {

  private query: string;

  private lang: string;

  private screenName: string;

  private until: string;


  get _query(): string {
    return this.query;
  }

  set _query(value: string) {
    this.query = value;
  }

  get _lang(): string {
    return this.lang;
  }

  set _lang(value: string) {
    this.lang = value;
  }

  get _screenName(): string {
    return this.screenName;
  }

  set _screenName(value: string) {
    this.screenName = value;
  }

  get _until(): string {
    return this.until;
  }

  set _until(value: string) {
    this.until = value;
  }
}
